### PR TITLE
[netcore] Fix marshalling of ref ptr return types

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: mono/metadata/object.c [hunk 1]
```diff
@@ -5566,12 +5566,22 @@ mono_runtime_try_invoke_array (MonoMethod *method, void *obj, MonoArray *params,
...
+				tmpret->byref = FALSE;
+				box_args [1] = mono_type_get_object_checked (mono_domain_get (), tmpret, error);
+				mono_metadata_free_type (tmpret);
+			} else {
+				box_args [1] = mono_type_get_object_checked (mono_domain_get (), sig->ret, error);
+			}
 			return_val_if_nok (error, NULL);
 
 			res = mono_runtime_try_invoke (box_method, NULL, box_args, &box_exc, error);
-			g_assert (box_exc == NULL);
```
Comment (vargaz) [inline mono/metadata/object.c]: Why is this needed ?

Comment (filipnavara) [reply mono/metadata/object.c]: It's not, but it was easier to debug a managed exception than a runtime assert with no message. I can revert this part of necessary.

Comment (vargaz) [reply mono/metadata/object.c]: I think we assert because that method is not supposed to fail.

Comment (filipnavara) [reply mono/metadata/object.c]: It fails only when something is internally inconsistent. I'll revert this one back to the assert.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by filipnavara** [comment]
@monojenkins build Linux AArch64 FullAOT+Interpreter
